rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -175,10 +175,19 @@ TEST_F(RecordIntegrationTestFixture, qos_is_stored_in_metadata)
   EXPECT_THAT(recorded_messages, SizeIs(expected_messages));
 
   auto recorded_topics = mock_writer.get_topics();
-  std::string serialized_profiles = recorded_topics.at(topic).first.offered_qos_profiles;
+  auto offered_qos_profiles = recorded_topics.at(topic).first.offered_qos_profiles;
+
+
+  std::vector<rosbag2_storage::Rosbag2QoS> to_encode;
+  to_encode.reserve(offered_qos_profiles.size());
+  std::transform(
+    offered_qos_profiles.begin(), offered_qos_profiles.end(), std::back_inserter(to_encode),
+    [](auto & qos) {return static_cast<rosbag2_storage::Rosbag2QoS>(qos);});
+  auto temp = YAML::convert<std::vector<rosbag2_storage::Rosbag2QoS>>::encode(to_encode);
+  std::string serialized_profiles = YAML::Dump(temp);
```

Comment:
Here, a conversion from `std::vector<rclcpp::QoS>` & serialized YAML (`std::string`) is needed.